nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -502,69 +556,103 @@ def quantize_weight(self, wrapper, **kwargs):
         return weight
 
     def quantize_input(self, inputs, wrapper, **kwargs):
-        config = wrapper.config
         module = wrapper.module
-        input_bits = get_bits_length(config, 'input')
 
-        module.input_bit = torch.tensor([input_bits])
-        quant_start_step = config.get('quant_start_step', 0)
-        assert input_bits >= 1, "quant bits length should be at least 1"
+        layer_quant_setting = module.layer_quant_setting
+        tensor_quant_setting = layer_quant_setting.input
 
-        if quant_start_step > int(self.bound_model.steps):
-            current_min, current_max = torch.min(inputs), torch.max(inputs)
-            module.tracked_min_input.copy_(current_min)
-            module.tracked_max_input.copy_(current_max)
+        # layer-wise settings
+        quant_start_step = layer_quant_setting.quant_start_step
+        ema_decay = layer_quant_setting.ema_decay
+
+        # tensor-wise settings
+        dtype = tensor_quant_setting.quant_dtype
+        scheme = tensor_quant_setting.quant_scheme
+        qmin, qmax = tensor_quant_setting.get_qmin_qmax()
+        bits = tensor_quant_setting.bits
+
+        if not wrapper.training:
+            scale = module.input_scale
+            zero_point = module.input_zero_point
+            inputs = self._quantize(inputs, scale, zero_point, qmin, qmax)
+            inputs = self._dequantize(inputs, scale, zero_point)
             return inputs
 
-        # we dont update output quantization parameters in evaluation stage
-        if wrapper.training:
-            current_min, current_max = torch.min(inputs), torch.max(inputs)
-            current_min = update_ema(module.tracked_min_input, current_min, module.ema_decay)
-            current_max = update_ema(module.tracked_max_input, current_max, module.ema_decay)
+        current_min, current_max = get_min_max_value(inputs, QuantType.INPUT, scheme)
+
+        if int(self.bound_model.steps) == 1:
             module.tracked_min_input.copy_(current_min)
             module.tracked_max_input.copy_(current_max)
 
+        tracked_min_input = update_ema(module.tracked_min_input, current_min, ema_decay)
+        tracked_max_input = update_ema(module.tracked_max_input, current_max, ema_decay)
+        module.tracked_min_input.copy_(tracked_min_input)
+        module.tracked_max_input.copy_(tracked_max_input)
+
+        if quant_start_step > int(self.bound_model.steps):
```

Comment:
Why delete original logic which can help initiate `tracked_min_input` and `tracked_max_input` here? According to the test result before, keeping them can perform better or converge faster than removing them.

Reply:
I see, it will be kept whatever `quant_start_step > int(self.bound_model.steps)` or not.